integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java

```diff
@@ -0,0 +1,692 @@
+
+        mock.assertIsSatisfied();
+
+    }
+
+    @Path("jackson/marshal-includedefault")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public String jacksonMarshalIncludeDefalut() throws Exception {
+        TestOtherPojo pojo = new TestOtherPojo();
+        pojo.setName("Camel");
+        return producerTemplate.requestBody("direct:jackson-marshal-includedefault-marshal",
+                pojo,
+                String.class);
+    }
+
+    @Path("jackson/unmarshal-array")
+    @GET
+    public void jacksonUnmarshalArray(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-endArray", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(String[].class);
+
+        String json = "[\"Camel\", \"World\"]";
+        producerTemplate.sendBody("direct:jackson-unmarshal-beginArray", json);
+
+        mock.assertIsSatisfied();
+
+        String[] array = mock.getReceivedExchanges().get(0).getIn().getBody(String[].class);
+        assertNotNull(array);
+        assertEquals(2, array.length);
+
+        String string = array[0];
+        assertEquals("Camel", string);
+        string = array[1];
+        assertEquals("World", string);
+    }
+
+    @Path("jackson/marshal-contenttype-header")
+    @GET
+    public void jacksonMarshalContentTypeHeader() throws Exception {
+        final Map<String, Object> in = new HashMap<>();
+        in.put("name", "Camel");
+
+        Exchange out = producerTemplate.request("direct:jackson-marshal-ct-yes", exchange -> exchange.getIn().setBody(in));
+
+        assertNotNull(out);
+        assertTrue(out.hasOut());
+        assertEquals("application/json", out.getMessage().getHeader(Exchange.CONTENT_TYPE));
+        out = producerTemplate.request("direct:jackson-marshal-ct-yes2", exchange -> exchange.getIn().setBody(in));
+
+        assertNotNull(out);
+        assertTrue(out.hasOut());
+        assertEquals("application/json", out.getMessage().getHeader(Exchange.CONTENT_TYPE));
+
+        out = producerTemplate.request("direct:jackson-marshal-ct-no", exchange -> exchange.getIn().setBody(in));
+
+        assertNotNull(out);
+        assertTrue(out.hasOut());
+        assertNull(out.getMessage().getHeader(Exchange.CONTENT_TYPE));
+    }
+
+    @Path("jackson/marshal-general")
+    @GET
+    public void jacksonMarshalGeneral() throws Exception {
+        Map<String, Object> in = new HashMap<>();
+        in.put("name", "Camel");
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-marshal-reverse", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(Map.class);
+        mock.message(0).body().isEqualTo(in);
+
+        Object marshalled = producerTemplate.requestBody("direct:jackson-marshal-in", in);
+        String marshalledAsString = context.getTypeConverter().convertTo(String.class, marshalled);
```

Comment:
Sometimes, I uses requestBody(...., ..., String.class)

Reply:
Good to know! I just use the way how it is tested in Camel